Doc/library/typing.rst

```diff
@@ -849,6 +849,22 @@ using ``[]``.
       concat(b"foo", b"bar")  # OK, output has type 'bytes'
       concat("foo", b"bar")   # Error, cannot mix str and bytes
 
+   Note that ``AnyStr`` and ``str | bytes`` are different from each other and
+   have different use cases. ``str | bytes`` does allow you to mix str and bytes::
+
+      def concat(a: str | bytes, b: str | bytes) -> str | bytes:
+          return a + b
+
+      concat("foo", b"bar")  # Passes typecheck but raises an error at runtime
```

Comment:
Hmm, the usability issue I see is that people are using `AnyStr` _too much_ -- they often use it when they _should_ be using `str | bytes`. But this is a good example of a place where `AnyStr` is _appropriate_. Maybe we should skip this example and just use your `greet_bad` and `greet_proper` examples below, which are quite good?